### PR TITLE
Basic combat

### PR DESCRIPTION
**Pull request by 0x72D0**
WIP for #57 

For now I have a simple combat system. The character simply move to the target and the game call a function for the currently selected unit and this function call at each strike a receive damage function to the target. It select the first attack only for now

For now there is a lot of bug I need to fix, but let me know what do you think about this implementation and how can I improve it :)

### CODE REVIEW

File: source/game/Game.gd [hunk 1]
```diff
@@ -29,6 +29,12 @@ func _unhandled_input(event: InputEvent) -> void:
 			elif current_unit and not loc.unit:
 				current_unit.move_to(_get_path_for_unit(current_unit, loc))
 				_set_current_unit(null)
+			
+			elif current_unit and loc.unit and loc.unit.side.number != current_side.number:
+				current_unit.move_to(_get_path_for_unit(current_unit, loc))
```
Comment (GaidamakUA) [inline source/game/Game.gd]: Won't it move unit even if enemy unit is out of reach?

**Comment by GaidamakUA** [comment]
We also need a way to tell from which side to attack.

**Comment by GaidamakUA** [comment]
I think your code will already work like that. I meant we need a way to choose direction from which to attack an enemy.
In the first one is it possible to attack enemy unit if it's tile is unreachable but it's right next to reachable area?
I think, we need to store tile from which mouse pointer entered enemy tile. If it's reachable that it's possible to attack enemy unit. And in future we would add arrow pointing from that hex to enemy unit hex.

**Comment by 0x72D0** [comment]
@GaidamakUA Maybe this could have is own issue and pull request? They are a lot to cover for this to happen. We have to detect the direction of the mouse when it's hovering a unit sprite and we need to adapt the pathfinding algorithm.

**Comment by 0x72D0** [comment]
I just finish to add the interface to allow the user to choose which attack he want to perform

**Comment by Byteron** [comment]
I like what's happening here! :D
minor request: 
If you're still working on this, mind marking it with a [WIP] in the title?
Thanks!

File: source/game/Game.gd [hunk 2]
```diff
@@ -29,6 +30,17 @@ func _unhandled_input(event: InputEvent) -> void:
...
 				current_unit.move_to(_get_path_for_unit(current_unit, loc))
 				_set_current_unit(null)
+			
+			elif (current_unit and loc.unit and loc.unit.side.number != current_side.number  
+				  and current_unit.reachable.has(loc)):
+				
+				current_unit.move_to(_get_path_for_unit(current_unit, loc))
+				yield(scenario, "unit_move_finished")
+				HUD.show_attack_popup(current_unit)
+				yield(HUD, "attack_selected")
```
Comment (Byteron) [inline source/game/Game.gd]: I'm pretty sure you can move `current_unit.execute_attack(loc.unit, current_attack)` and `_set_current_unit(null)` into the `_on_attack_selected"` function, then you don't need to `yield(HUD, "attack_selected")` *and* you don't need to store the attack in an additional member variable (so `var current_attack` can be removed)

Comment (0x72D0) [reply source/game/Game.gd]: Nice, thanks for the feedback I change my code to delete the yield and the current_attack variable.

**Comment by 0x72D0** [comment]
Ok I think this PR is ready to be reviewed and pull :)

**Comment by Byteron** [comment]
merged.
it's a good start, thanks. :)

**Pull request merged**